fhir-validation/src/main/java/com/ibm/watson/health/fhir/validation/util/ProfileSupport.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * (C) Copyright IBM Corp. 2019
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.watson.health.fhir.validation.util;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.ibm.watson.health.fhir.model.annotation.Constraint;
+import com.ibm.watson.health.fhir.model.resource.Resource;
+import com.ibm.watson.health.fhir.model.resource.StructureDefinition;
+import com.ibm.watson.health.fhir.model.type.Canonical;
+import com.ibm.watson.health.fhir.model.type.ElementDefinition;
+import com.ibm.watson.health.fhir.model.type.Meta;
+import com.ibm.watson.health.fhir.model.type.TypeDerivationRule;
+import com.ibm.watson.health.fhir.registry.FHIRRegistry;
+
+public final class ProfileSupport {
```

Comment:
Why final?

Reply:
So that it can’t be extended. Good practice for utility classes along with a private constructor.